app/controllers/deregister/DeregistrationCheckYourAnswersController.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.deregister
+
+import cats.implicits._
+import connectors.{AmlsConnector, DataCacheConnector}
+import controllers.{AmlsBaseController, CommonPlayDependencies}
+import models.businessmatching.BusinessActivity.HighValueDealing
+import models.businessmatching.BusinessMatching
+import models.deregister.{DeRegisterSubscriptionRequest, DeregistrationReason}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.AuthEnrolmentsService
+import services.cache.Cache
+import uk.gov.hmrc.mongo.play.json.Codecs
+import utils.{AckRefGenerator, AuthAction}
+import views.html.deregister.DeregistrationCheckYourAnswersView
+
+import java.time.LocalDate
+import javax.inject.Inject
+
+class DeregistrationCheckYourAnswersController @Inject()(authAction: AuthAction,
+                                                         ds: CommonPlayDependencies,
+                                                         dataCacheConnector: DataCacheConnector,
+                                                         amlsConnector: AmlsConnector,
+                                                         authEnrolmentsService: AuthEnrolmentsService,
+                                                         cc: MessagesControllerComponents,
+                                                         view: DeregistrationCheckYourAnswersView) extends AmlsBaseController(ds, cc) {
+
+  def get: Action[AnyContent] = authAction.async {
+    implicit request =>
+      dataCacheConnector
+        .fetchAll(request.credId)
+        .map(_.getOrElse(throw new RuntimeException("missing 'Cache' in mongo")))
```

Comment:
I understand the thought process of throwing when there is no data retrieved from the cache, however for the scope of this work, I agree with Lewis and believe it would be better to maintain consistent behaviour across the service. We could potentially raise a ticket to review this behaviour (with design team included)?